x-pack/plugins/ml/public/application/data_frame_analytics/pages/analytics_creation/components/validation_step/validation_step_wrapper.tsx

```diff
@@ -5,46 +5,129 @@
  * 2.0.
  */
 
-import React, { FC, useState } from 'react';
+import React, { FC, useEffect, useMemo, useState } from 'react';
 import { EuiForm } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { debounce } from 'lodash';
 
 import { CreateAnalyticsStepProps } from '../../../analytics_management/hooks/use_create_analytics_form';
 import { ValidationStep } from './validation_step';
 import { ValidationStepDetails } from './validation_step_details';
 import { ANALYTICS_STEPS } from '../../page';
+import { useMlApiContext } from '../../../../../contexts/kibana';
+import { getJobConfigFromFormState } from '../../../analytics_management/hooks/use_create_analytics_form/state';
+import { extractErrorMessage } from '../../../../../../../common/util/errors';
+import {
+  CalloutMessage,
+  ValidateAnalyticsJobResponse,
+  VALIDATION_STATUS,
+} from '../../../../../../../common/constants/validation';
 
 export interface ValidationSummary {
   warning: number;
   success: number;
 }
 
 export const ValidationStepWrapper: FC<CreateAnalyticsStepProps> = ({
-  actions,
   state,
   setCurrentStep,
   step,
   stepActivated,
 }) => {
-  const [validationSummary, setValidationSummary] = useState<ValidationSummary>({
-    warning: 0,
-    success: 0,
-  });
-
+  const [checksInProgress, setChecksInProgress] = useState<boolean>(false);
+  const [validationMessages, setValidationMessages] = useState<CalloutMessage[]>([]);
+  const [errorMessage, setErrorMessage] = useState<CalloutMessage | undefined>();
+  const { form, jobConfig, isAdvancedEditorEnabled } = state;
+  const {
+    dependentVariable,
+    jobType,
+    trainingPercent,
+    numTopFeatureImportanceValues,
+    numTopClasses,
+    includes,
+  } = form;
   const showValidationStep = step === ANALYTICS_STEPS.VALIDATION;
   const showDetails = step !== ANALYTICS_STEPS.VALIDATION && stepActivated === true;
+  const {
+    dataFrameAnalytics: { validateDataFrameAnalytics },
+  } = useMlApiContext();
 
   const dataTestSubj = `mlAnalyticsCreateJobWizardValidationStepWrapper${
     showValidationStep ? ' active' : ''
   }${showDetails ? ' summary' : ''}`;
 
+  const debouncedValidationChecks = debounce(async () => {
```

Comment:
It will create a new callback on each render. I'd recommend using [useDebounce](https://github.com/streamich/react-use/blob/master/docs/useDebounce.md) instead. To import it use `import useDebounce from 'react-use/lib/useDebounce';`